syntax = "proto3";

// clang-format off

package rips;

service WellPath {
  rpc GetCompletionData(SimulatorTableRequest)
      returns (SimulatorTableData) {}
}

message SimulatorTableRequest {
  string wellpath_name = 1;
  int32 case_id = 2;
}

message SimulatorCompdatEntry {
  string well_name = 1;
  int32 grid_i = 2;
  int32 grid_j = 3;
  int32 upper_k = 4;
  int32 lower_k = 5;
  string open_shut_flag = 6;
  optional double saturation = 7;
  optional double transmissibility = 8;
  optional double diameter = 9;
  optional double kh = 10;
  optional double skin_factor = 11;
  optional double d_factor = 12;
  optional string direction = 13;
  optional double start_md = 14;
  optional double end_md = 15;
  optional string comment = 16;
}

message SimulatorWelspecsEntry {
  string well_name = 1;
  string group_name = 2;
  int32 grid_i = 3;
  int32 grid_j = 4;
  optional double bhp_depth = 5;
  string phase = 6;
  optional double drainage_radius = 7;
  optional string inflow_equation = 8;
  optional string auto_shut_in = 9;
  optional string cross_flow = 10;
  optional int32 pvt_num = 11;
  optional string hydrostatic_density_calc = 12;
  optional int32 fip_region = 13;
}

message SimulatorWelsegsHeaderEntry {
  string well_name = 1;
  double top_depth = 2;
  double top_length = 3;
  optional double wellbore_volume = 4;
  string info_type = 5;
  optional string pressure_omponents = 6;
  optional string flow_model = 7;
}

message SimulatorWelsegsRowEntry {
  int32 segment_1 = 1;
  int32 segment_2 = 2;
  int32 branch = 3;
  int32 join_segment = 4;
  double length = 5;
  double depth = 6;
  optional double diameter = 7;
  optional double roughness = 8;
  optional string description = 9;
}

message SimulatorWelsegsEntry {
  SimulatorWelsegsHeaderEntry header = 1;
  repeated SimulatorWelsegsRowEntry row = 2;
}

message SimulatorCompsegsRowEntry {
  int32 i = 1;
  int32 j = 2;
  int32 k = 3;
  int32 branch = 4;
  double distance_start = 5;
  double distance_end = 6;
  optional string grid_name = 7;
}

message SimulatorCompsegsEntry {
  repeated SimulatorCompsegsRowEntry rows = 1;
}

message SimulatorTableData {
  repeated SimulatorCompdatEntry compdat = 1;
  repeated SimulatorWelspecsEntry welspecs = 2;
  repeated SimulatorWelsegsEntry welsegs = 3;
  repeated SimulatorCompsegsEntry compsegs = 4;
}

